### DISCUSSION TITLE
Downgrades aren't allowed in a <Bundle> but they are in a <Package> MSI?

### DISCUSSION THREAD

**Discussion by george7378** [wixtoolset/issues#8792] [Questions]
Hi all - I'm trying to allow downgrades in my Wix project, so that older versions can be properly installed on top of newer versions.

I've tried to do it by adding this to my `<Package>`:

```
<MajorUpgrade AllowDowngrades="yes" />
<Property Id="REINSTALLMODE" Value="dmus" /> <!-- I guess amus would work too, and seems to be the more common choice -->

<!-- Another alternative seems to be doing this:
<InstallExecuteSequence>
  <RemoveExistingProducts Overridable="yes" Before="CostInitialize" />
</InstallExecuteSequence>
-->
```

This seems to compile fine. However the project is wrapped in a `<Bundle>` bootstrapper, and I can't find any way to enable downgrades in this. Discussions seem to show that it can't be done for Wix bootstrappers.

Is this true - and if so, why would it be allowed by MSIs and not by bundles?

I did try `SuppressDowngradeFailure="true"` in combination with my package changes above, but it still fails (only it shows the 'success' page instead of the error code).

Thanks for your help!

**Comment by george7378** [comment]
Wondering if anyone has some input on this? I guess it's because in a bundle, it's possible that one of the contained installers might not allow downgrades, so the bundle itself can't either?

**Reply by robmen** [reply]
It's possible. We (FireGiant) did it for a customer a long time ago. I don't remember the details. I'm pretty sure a custom BA was required.

**Reply by george7378** [reply]
Ah right, thanks for the info :) In our installer we're bundling .NET runtimes which seem also to be built with WiX, and don't allow downgrades, so it seems it wouldn't work anyway.
